### PR TITLE
Refactored the resampling scripts 

### PR DESCRIPTION
**Pull request by langer-net**

### CODE REVIEW

File: src/resampling_crop_type.py [hunk 1]
```diff
@@ -19,71 +24,129 @@ def __init__(self, ROOT_DIR:str) -> None:
...
+        ) -> pd.DataFrame:
+        # rank the labels by their count
+        label_ranking = df_label_comb[["label", "field_id"]] \
+            .groupby("label").count().sort_values("field_id").reset_index()
+        return label_ranking
+    
+    def get_base_label_index(self, label_ranking:pd.DataFrame
+        ) -> int:
+        # get the index and label of a label in the middle of the ranking (the base label)
+        sampling_base_label_idx = int((len(label_ranking)/2) - 1)
```
Comment (Fbehr-data) [inline src/resampling_crop_type.py:49]: median?

File: src/resampling_crop_type.py [hunk 2]
```diff
@@ -19,71 +24,129 @@ def __init__(self, ROOT_DIR:str) -> None:
...
+        sampling_size = int(label_ranking.iloc[sampling_base_label_idx]["field_id"])
+        return sampling_size 
 
-    def save_train_test_data(self):
+    def get_labels_to_resample(self, label_ranking:pd.DataFrame
+        ) -> Union[pd.Series, pd.Series]:
+        # get the classes for which we need to do over- and undersampling
+        sampling_base_label_idx = self.get_base_label_index(label_ranking) 
+        classes_to_oversample = label_ranking[0:sampling_base_label_idx]["label"]
+        classes_to_undersample = label_ranking[sampling_base_label_idx+1:len(label_ranking)]["label"]
```
Comment (Fbehr-data) [inline src/resampling_crop_type.py:72]: transform this label_ranking[sampling_base_label_idx+1:len(label_ranking)]["label"]

to 

 label_ranking[sampling_base_label_idx+1:-1]["label"]

File: src/resampling_crop_type.py [hunk 3]
```diff
@@ -19,71 +24,129 @@ def __init__(self, ROOT_DIR:str) -> None:
...
+        # set the resampling method
+        if oversampling:
+            resampler = RandomOverSampler(sampling_strategy=strategy)
+            method = "oversampled"
+        else:
+            resampler = RandomUnderSampler(sampling_strategy=strategy)
+            method = "undersampled" 
+        # start the resampling process
+        X_resampled, y_resampled = resampler.fit_resample(X_resampling, y_resampling)
+        print(f"{method} labels: {Counter(y_resampled)}")
```
Comment (Fbehr-data) [inline src/resampling_crop_type.py:91]: print Out values before under or Oversampling???

**Review by Fbehr-data — COMMENTED** [review]
Look at the comments

File: src/resampling_crop_type.py (hunk 3 above)
Comment (langer-net) [reply src/resampling_crop_type.py:91]: Added the second callout before the resampling.

**Pull request merged**